File: interview/01.02.go
/*
 * @Author: lmio
 * @Date: 2023-04-10 23:31:29
 * @LastEditTime: 2023-04-10 23:35:21
 * @FilePath: /leetcode/interview/01.02.go
 * @Description:面试题 01.02. 判定是否互为字符重排
 */
package interview

func CheckPermutation(s1 string, s2 string) bool {
	code := make([]int, 26)
	if len(s1) != len(s2) {
		return false
	}
	for i := range s1 {
		code[s1[i]-'a']++
	}
	for i := range s2 {
		if code[s2[i]-'a'] <= 0 {
			return false
		}
		code[s2[i]-'a']--
	}
	return true
}

File: interview/01.02.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-08-31 19:30:45
LastEditors: lmio 2091319361@qq.com
Description: 面试题 01.02. 判定是否互为字符重排
'''

from collections import Counter


class Solution:
    def CheckPermutation(self, s1: str, s2: str) -> bool:
        s1 = Counter(s1)
        s2 = Counter(s2)
        return s1 == s2